kie-server-parent/kie-server-controller/kie-server-controller-impl/src/main/java/org/kie/server/controller/impl/KieServerHealthCheckControllerImpl.java

```diff
@@ -0,0 +1,137 @@
+import javax.annotation.PreDestroy;
+
+import org.kie.server.api.model.KieServerInfo;
+import org.kie.server.controller.api.ModelFactory;
+import org.kie.server.controller.api.model.runtime.ServerInstanceKey;
+import org.kie.server.controller.api.model.spec.ServerTemplate;
+import org.kie.server.controller.api.model.spec.ServerTemplateKey;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class KieServerHealthCheckControllerImpl extends KieServerControllerImpl {
+
+    private static final String PING_ALIVE_PROPERTY = "kie.ping.alive.timeout";
+
+    private static final String PING_ALIVE_DISABLED = "kie.ping.alive.disable";
+
+    private static long PING_INTERVAL;
+
+    private static boolean PING_DISABLED;
+
+    private static final Logger logger = LoggerFactory.getLogger(KieServerHealthCheckControllerImpl.class);
+
+    private Ping ping;
+
+    private ExecutorService executorService;
+
+    static {
+        try {
+            PING_INTERVAL = Long.parseLong(System.getProperty("kie.ping.alive.timeout", "5000"));
+        } catch (NumberFormatException e) {
+            logger.warn("The property " + PING_ALIVE_PROPERTY + " is not a number; Fallback to 5000 ms ping");
+            PING_INTERVAL = 5000L;
+        }
+        try {
+            PING_DISABLED = Boolean.parseBoolean(System.getProperty("kie.ping.alive.disable", "false"));
+        } catch (NumberFormatException e) {
+            logger.warn("The property " + PING_ALIVE_DISABLED + " is not true/false value; Fallback to false");
+            PING_DISABLED = false;
+        }
+    }
+
+    public void setExecutorService(ExecutorService executorService) {
+        this.executorService = executorService;
+    }
+
+    class Ping implements Runnable {
+
+        private AtomicBoolean stop = new AtomicBoolean(false);
+
+        @Override
+        public void run() {
+            try {
+                while (!stop.get()) {
+                    for (KieServerInfo kieServerInfo : getServerInfoList()) {
+                        ServerInstanceKey serverInstanceKey = ModelFactory.newServerInstanceKey(kieServerInfo.getServerId(), kieServerInfo.getLocation());
+                        if (!KieServerInstanceManager.getInstance().isAlive(serverInstanceKey)) {
+                            logger.debug("ping isAlive " + kieServerInfo.getLocation() + ": KO. disconnected.");
+                            disconnect(kieServerInfo);
+                        } else {
+                            logger.debug("ping isAlive " + kieServerInfo.getLocation() + ": OK");
+                        }
+                    }
+                    Thread.sleep(PING_INTERVAL);
+                }
+            } catch (InterruptedException e) {
+                logger.warn("Rest Kie health check was interrupted");
+            }
+
+        }
+
+        public void stop() {
+            stop.set(true);
+        }
+
+    }
+
+    @PostConstruct
+    public synchronized void start() {
```

Comment:
I guess these @PostConstruct is not really used since there is servlet listener responsible for init if the ping, same goes for @PreDestroy